### ISSUE TITLE
Font caching Issue for ios

### ISSUE THREAD

**Issue by nibblis** [icerockdev/moko-resources#712]
In my project, fonts consume a significant amount of memory when launched on ios devices. There are no issues on android.

According to the [FontResource](https://github.com/icerockdev/moko-resources/blob/master/resources-compose/src/iosMain/kotlin/dev/icerock/moko/resources/compose/FontResource.ios.kt) code, fonts should be cached.

In my project, I specify a list of all fonts (18 in total) in a map:
```
@Composable
fun MontserratFontFamily() = mapOf(
    Montserrat.Black to fontFamilyResource(MR.fonts.montserrat_black.montserrat_black),
    Montserrat.Light to fontFamilyResource(MR.fonts.montserrat_light.montserrat_light),
    Montserrat.Bold to fontFamilyResource(MR.fonts.montserrat_bold.montserrat_bold),
    ...
)
```

Without fonts, the app usually takes up 150-200 MB of memory. With fonts, it takes 1.5-2 GB.
<img width="935" alt="2024-05-17 " src="https://github.com/icerockdev/moko-resources/assets/25844665/b6858b53-952f-447d-b26e-591558da3790">

Here is a screenshot from the profiler showing 3000 entries for fonts:
<img width="1332" alt="2024-05-17" src="https://github.com/icerockdev/moko-resources/assets/25844665/e14612ce-9753-4c5f-be40-c6f3482b019d">

**Comment by ExNDY** [comment]
Try to use:
https://github.com/Kotlin/kotlinx.collections.immutable
map is not Immutable type, i think on android you doesn't have problem because in latest versions of jetpack compose has optimisations with copy this library under hood(or you have problem on android, but don't see it). My opinion you should use persistentMapOf, not simple map.

**Comment by nibblis** [comment]
@ExNDY, it's the same with `persistentMapOf`
according to the profiler in android, the memory usage does not exceed 160 MB
<img width="732" alt="2024-05-20" src="https://github.com/icerockdev/moko-resources/assets/25844665/de59e974-e22a-4a9b-8c99-79e52df7583f">

**Comment by ExNDY** [comment]
Ios != Android, 
for android used jetpack compose, for ios jetbrains solution. U should think about recomposition and ios platform is sensitive for this.
Please try on ios and say your result

**Comment by nibblis** [comment]
not sure what exactly you mean by recomposition

i tried copying the `fontFamilyResource` code to my project: removed `@Composable` and `remember`, made the fonts a constant. after that, memory consumption does not exceed 200 MB

i don't think the problem is with the moko-resources library

```
val montserratFontFamilyMap: Map<Montserrat, FontFamily> by lazy {
    mapOf(
    Montserrat.Black to fontFamilyResource(MR.fonts.montserrat_black.montserrat_black),
    Montserrat.Light to fontFamilyResource(MR.fonts.montserrat_light.montserrat_light),
    Montserrat.Bold to fontFamilyResource(MR.fonts.montserrat_bold.montserrat_bold),
    ...
    )
)
```

**Comment by ExNDY** [comment]
i don't cleary understand you, you find problem with your logic on ui? or what's happined? Recomposition for me is base of compose:) If you use don't stable types inside composition on every recomposition of screen (recreating/update screen content you create new values for unstable variable) because one of solution for best performance don't usage of list for remember, for example.
Because i'm question for you, about usage in yours code unstable map.
If your problem is solved - is cool.

**Comment by nibblis** [comment]
i haven't solved the problem, most likely it's because of `remember` and how it's implemented for ios

if you think i'm using unstable types, then why isn't there this problem on android?

**Comment by ExNDY** [comment]
you read my first message? Android has optimisations under hood. How much experience you have in compose? I need understand what i can do for this issue

**Comment by nibblis** [comment]
yes, i saw the message. but your assumption that android has optimization under the hood doesn't seem convincing. simply removing `remember` eliminates the problem

fonts don't depend on the ui state. is it possible to make `fontFamilyResource` not `@Composable`?

**Comment by ExNDY** [comment]
Hm, man, you "rewrite" on non compose style. You don't hear important argument: jetpack compose=\=compose Multiplatform on other platforms, that framework very sensitive for data types. Usage of lazy inside ```
@Composable
``` talking about you don't know principle of Compose. Sorry, i'm question you about your experience in Compose for that. You ignore me with this question. Question about persistentMapOf on ios was ignored too. If somebody wants chat with you that's cool.
